typing/typedtree.mli

```diff
@@ -487,6 +489,7 @@ and value_binding =
   {
     vb_pat: pattern;
     vb_expr: expression;
+    vb_sort: Types.sort;
```

Comment:
What if a value is immediate?

Reply:
The `sort` is sufficient to know how to compile the pattern match (and in fact, after discussion with Leo, I plan to change the other layouts in the typed tree to sorts, defaulted to value for `any` cases, as that's enough information).

Reply:
Hm. Interesting. That makes sense. So, for any situation that can actually be compiled (that is, not `any`), all we need is the sort. That sounds right. Still, we do certain optimizations over immediates, and so storing only the sort -- while sufficient -- might miss these optimizations. My very uneducated guess is that the immediate-optimizations (which pre-date layouts, of course) look elsewhere. Maybe that's all OK, but I wonder if there's an opportunity for consolidation. In any case, I agree that `sort` works here.